src/slackeros/ros_connector.py

```diff
@@ -95,6 +107,69 @@ def __init__(
         self.queue_worker.setDaemon(True)
         self.queue_worker.start()
 
+    def log_image(self, type=None, topic='/head_xtion/rgb/image_color'):
+        class ImageUploader(Thread):
+
+            def __init__(self, access_token, bridge, send_image, type=None, image_format='jpeg', whitelist_channels=[], image_upload_title="", topic='/head_xtion/rgb/image_color'):
+                Thread.__init__(self)
+                self.access_token = access_token
+                self.bridge = bridge
+                self.send_image = send_image
+                self.type = type
+                self.image_format = image_format
+                self.whitelist_channels = whitelist_channels
+                self.image_upload_title = image_upload_title
+                self.topic = topic
+
+            def run(self):
+                # upload image
+                try:
+                    image = rospy.wait_for_message(self.topic, Image, timeout=1.5)
+                except rospy.ROSException as e:
+                    rospy.logwarn("No image retrieved before timeout")
+                    return
+
+                if self.type is None:
+                    self.type = image.encoding
+                try:
+                    # Convert your ROS Image message to OpenCV2
+                    if self.type == "rgb8": self.type = "bgr8"
+                    cv2_img = self.bridge.imgmsg_to_cv2(image, self.type)
+                except CvBridgeError, e:
+                    rospy.logwarn(e)
+                    return
+
+                # Save your OpenCV2 image
+                image_file, image_path = tempfile.mkstemp('.' + self.image_format)
+                try:
+                    cv2.imwrite(image_path, cv2_img)
+                except Exception as e:
+                    rospy.logwarn("Exception writing image to file %s" % image_path)
+                    return
+
+                # upload to slack
+                params = {
+                    'token': self.access_token,
+                    'channels': self.whitelist_channels,
```

Comment:
does this now mean it will be sent to _all_ the white-list channels?

Reply:
actually now this `whitelist_channels` is left empty, so apparently i could just rename it. Where do you set the channel for sending messages?

Reply:
`whitelist_channel` is used to only allow people on these channels to submit `\` slash commands. So, it's something entirely different. But the image should be uploaded to the channel that is configured for the incoming hook, right? That is not configured in the app yet, as it is currently configured on slack itself. For the `\rostopic poll` slash command, obviously it should be uploaded to the channel the command was issued on.

Reply:
but with images I am not using a webhook  and I need to have a list of channels. For when `/rostopic poll` is issued I can take the channel from the payload msg, but how do I take the channel configured with the webhook?

Reply:
Exactly, that's the problem. It's not ideal, but you'll have to give it as an argument at start time, I think.